sparkmagic/sparkmagic/livyclientlib/endpoint.py

```diff
@@ -1,14 +1,16 @@
 from .exceptions import BadUserDataException
+import sparkmagic.utils.constants as constants
 
 
 class Endpoint(object):
-    def __init__(self, url, username="", password=""):
+    def __init__(self, url, auth_type=constants.NO_AUTH, username="", password=""):
```

Comment:
Default should be None. If None, then get the value from a configuration called `default_authentication_method`.